src/NuGet.Core/NuGet.Packaging/PackageCreation/Extensions/FrameworksExtensions.cs

```diff
@@ -10,9 +10,14 @@ namespace NuGet.Packaging
     public static class FrameworksExtensions
     {
         // NuGet.Frameworks doesn't have the equivalent of the old VersionUtility.GetFrameworkString
-        // which is relevant for building packages
+        // which is relevant for building packages. This isn't needed for net5.0+ frameworks.
         public static string GetFrameworkString(this NuGetFramework self)
         {
+            if (self.Version.Major >= 5)
+            {
+                return self.DotNetFrameworkName;
```

Comment:
Isn't this supposed to be the short folder name?

Reply:
Nvm, dotnet framework name returns GetShortFolderName for .NET 5. These are confusing :(